tests/openfl/databases/test_tensor_db.py

```diff
@@ -226,3 +242,110 @@ def call(self, local_tensors, *_):
+            for _ in db_iterator:
+                pass  # TODO: to coverage _iterate()
+            tensors = [local_tensor.tensor for local_tensor in local_tensors]
+            return np.sum(tensors, axis=0)
+
+    tensor_key = TensorKey('tensor_name', 'agg', 0, False, ())
+
+    agg_nparray = tensor_db.get_aggregated_tensor(
+        tensor_key, collaborator_weight_dict, Sum())
+
+    assert np.array_equal(agg_nparray, np.array([2, 4, 6, 8, 10]))
+
+
+def test_retrieve(tensor_db):
+    """Test that TensorDB's retrieve method works correctly."""
+    ret = _retrieve(tensor_db.tensor_db, 'tensor_name', 'agg', 0, False, ('col1',))
+
+    assert np.array_equal(ret, np.array([0, 1, 2, 3, 4]))
+    # TODO: this test is enabled by a fix in dataframe.py:121
+
+
+def test_retrieve_no_result(tensor_db):
+    """Test that TensorDB's retrieve method retrieves none."""
+    ret = _retrieve(tensor_db.tensor_db)
+
+    assert ret is None
+    # TODO: this test is enabled by a fix in dataframe.py:121
+
+
+def test_search(tensor_db):
+    """Test that TensorDB's search method works correctly."""
+    ret = _search(tensor_db.tensor_db, 'tensor_name', 'agg', 0, False, ('col1',))
+
+    assert_frame_equal(ret, tensor_db.tensor_db.drop([1]))
+
+
+def test_search_no_result(tensor_db):
+    """Test that TensorDB's search method finds no result."""
+    ret = _search(tensor_db.tensor_db, 'tensor_name', 'agg', 1, False, ('col3',))
+
+    assert_frame_equal(ret, tensor_db.tensor_db)
+    # TODO: this test is enabled by a fix in dataframe.py:50
+
+
+def test_store(tensor_db):
+    """Test that TensorDB's store method works correctly."""
+    _store(tensor_db.tensor_db, 'tensor_name', 'agg', 0,
+           False, ('col1',), np.array([5, 6, 7, 8, 9]))
+    # TODO: this test is enabled by fixes in dataframe.py:90 & 98
+
+    assert np.array_equal(tensor_db.tensor_db.loc[0]['nparray'], np.array([5, 6, 7, 8, 9]))
+
+
+def test_store_no_nparray(tensor_db):
+    """Test that TensorDB's store method returns directly when no nparray provided."""
+    origin_tensor = tensor_db.tensor_db.copy(deep=True)
+    _store(tensor_db.tensor_db)
+
+    assert_frame_equal(origin_tensor, tensor_db.tensor_db)
+
+
+def test_store_not_overwrite(tensor_db):
+    """Test that TensorDB's store method changes nothing when disabling overwrite."""
+    origin_tensor = tensor_db.tensor_db.copy(deep=True)
+    _store(tensor_db.tensor_db, 'tensor_name', 'agg', 0, False,
+           ('col1',), np.array([5, 6, 7, 8, 9]), overwrite=False)
+    # TODO: this test is enabled by fixes in dataframe.py:90 & 98
+
+    assert_frame_equal(origin_tensor, tensor_db.tensor_db)
+
+
+def test_store_append_at_the_end(tensor_db):
+    """Test that TensorDB's store method appends new tensor at the end."""
+    _store(tensor_db.tensor_db, 'tensor_name', 'agg', 0, False,
+           ('col3',), np.array([5, 6, 7, 8, 9]))
+    # TODO: this test is enabled by fixes in dataframe.py:90 & 98
```

Comment:
@fangxiaoran can this TODO be removed given the comment?

Reply:
Yes, and yes for other "TODO"s.

I added these comments since some fixes were made to the source code. Should I make another commit to remove these comments?

Reply:
Yes please!

Reply:
Done by https://github.com/securefederatedai/openfl/pull/821/commits/9525e8516e61023f61a930a5507fdd6aa5c3f1c7.